### PR TITLE
make numeric and alphabetical backspace work.

### PR DESCRIPTION
**Pull request by this-mojtaba**

### CODE REVIEW

File: src/types/alphabetical.ts [hunk 1]
```diff
@@ -34,6 +35,12 @@ export class Alphabetical extends Interact {
 
       // TODO: implement backspace keyboard
       if (pressedKey.key === "backspace") {
+        if (input.length === 0) {
+          continue;
+        }
+        input = input.slice(0, -1);
+        moveLeft(1);
```
Comment (disizali) [inline src/types/alphabetical.ts]: await

File: src/types/numeric.ts [hunk 2]
```diff
@@ -31,6 +32,17 @@ export class Numeric extends Interact {
     ];
     for await (const pressedKey of readKeypress()) {
       if (typeof pressedKey.key == "undefined") continue;
+
+      if (pressedKey.key === "backspace") {
+        if (input.length === 0) {
+          continue;
+        }
+        input = input.slice(0, -1);
+        moveLeft(1);
```
Comment (disizali) [inline src/types/numeric.ts]: ditto

File: src/helpers/cursor.ts [hunk 3]
```diff
@@ -9,6 +9,7 @@ const DOWN = "B";
 const RIGHT = "C";
 const UP = "A";
 const CLEAR_DOWN = "0J";
+const CLEAR_RIGHT = "0P";
```
Comment (disizali) [inline src/helpers/cursor.ts]: use `0k` for clear until end of current line, please.

Comment (this-mojtaba) [reply src/helpers/cursor.ts]: of course @disizali dear.

**Comment by disizali** [comment]
Thanks, @Mojtaba-79 for this grate work!

**Review by disizali — APPROVED** [review]
LGTM\, Thank you @Mojtaba-79 for this great work!

**Pull request merged**